app/src/main/java/com/gh4a/activities/RepositoryActivity.java

```diff
@@ -374,19 +327,75 @@ protected Intent navigateUp() {
         return UserActivity.makeIntent(this, mRepoOwner);
     }
 
+    private void loadWatchingStatus(boolean refresh) {
+        RepositoryService.loadWatchingStatus(this, mRepoOwner, mRepoName, refresh)
+                .subscribe(result -> {
+                    mIsWatching = (Boolean) result;
+                    supportInvalidateOptionsMenu();
+                }, error -> {});
+    }
+
+    private void loadStarringStatus(boolean refresh) {
+        RepositoryService.loadStarringStatus(this, mRepoOwner, mRepoName, refresh)
+                .subscribe(result -> {
+                    mIsStarring = (Boolean)result;
+                    supportInvalidateOptionsMenu();
+                }, error -> {});
+    }
+
+    private void loadRepository(boolean refresh) {
+        Consumer<Repository> subscriber = repository -> {
+            mRepository = repository;
+            updateTitle();
+            invalidateTabs();
+            // Apply initial page selection first time the repo is loaded
+            if (mInitialPage >= PAGE_REPO_OVERVIEW && mInitialPage <= PAGE_ACTIVITY) {
+                getPager().setCurrentItem(mInitialPage);
+                mInitialPage = -1;
+            }
+            setContentShown(true);
+            supportInvalidateOptionsMenu();
+        };
+
+        RepositoryService.loadRepository(this, mRepoOwner, mRepoName, refresh)
+                .subscribe(subscriber, error -> {});
+    }
+
     @Override
     public boolean onOptionsItemSelected(MenuItem item) {
         String url = "https://github.com/" + mRepoOwner + "/" + mRepoName;
         switch (item.getItemId()) {
             case R.id.watch:
                 MenuItemCompat.setActionView(item, R.layout.ab_loading);
                 MenuItemCompat.expandActionView(item);
-                new UpdateWatchTask().schedule();
+
+                RepositoryService.setWatchingStatus(this, mRepoOwner, mRepoName, mIsWatching)
+                    .subscribe(o -> {
+                        if (mIsWatching == null) {
+                            // user refreshed while the action was in progress
+                            return;
+                        }
+                        mIsWatching = !mIsWatching;
+                        supportInvalidateOptionsMenu();
+                    }, error -> {});
                 return true;
             case R.id.star:
                 MenuItemCompat.setActionView(item, R.layout.ab_loading);
                 MenuItemCompat.expandActionView(item);
-                new UpdateStarTask().schedule();
+
+                RepositoryService.setStarringStatus(this, mRepoOwner, mRepoName, mIsStarring)
```

Comment:
Unchecked call warning. Probably another problem with generics. You should go over all your changes and make sure to fix all these warnings.